test/HaveLetLinter.lean

```diff
@@ -0,0 +1,144 @@
+import Mathlib.Tactic.Linter.HaveLetLinter
+import Mathlib.Tactic.Tauto
+
+/-- a tactic that simply logs an empty message.  Useful for testing the chattiness of the
+`haveLet` linter. -/
+elab "noise" : tactic => do Lean.logInfo ""
+
+set_option linter.haveLet 2 in
+#guard_msgs in
+-- check that `tauto`, `replace`, `classical` are ignored
+example : True := by
+  classical
+  let zero := 0
+  replace _zero := zero
+  let eq := (rfl : 0 = 0)
+  replace _eq := eq
+  tauto
+
+/--
+info:
+
+---
+warning: '_zero : Nat' is a Type and not a Prop. Consider using 'let' instead of 'have'.
+[linter.haveLet]
+-/
+#guard_msgs in
+example : True := by
+  noise
+  have ⟨_zero, _⟩ : Fin 1 := ⟨0, Nat.zero_lt_one⟩
+  exact .intro
+
+-- I don't know how to silence this test.
+--set_option linter.haveLet 2 in
+--/--
+--warning: '_zero : Nat' is a Type and not a Prop. Consider using 'let' instead of 'have'.
+--[linter.haveLet]
+---/
+--#guard_msgs in
+--example : True := by
+--  have ⟨_zero, _⟩ : Fin 1 := ⟨0, Nat.zero_lt_one⟩
+--  exact .intro
```

Comment:
```suggestion
/--
warning: '_zero : Nat' is a Type and not a Prop. Consider using 'let' instead of 'have'.
[linter.haveLet]
-/
#guard_msgs in
set_option linter.haveLet 2 in
example : True := by
  have ⟨_zero, _⟩ : Fin 1 := ⟨0, Nat.zero_lt_one⟩
  exact .intro
```

Is this what you want?